kotlinx-coroutines-core/common/src/channels/Channel.kt

```diff
@@ -23,12 +23,15 @@ import kotlin.jvm.*
  */
 public interface SendChannel<in E> {
     /**
-     * Returns `true` if this channel was closed by an invocation of [close]. This means that
-     * calling [send] will result in an exception.
+     * Returns `true` if this channel was closed by an invocation of [close] or its receiving side was [cancelled][ReceiveChannel.cancel].
+     * This means that calling [send] will result in an exception.
      *
-     * **Note: This is an experimental api.** This property may change its semantics and/or name in the future.
+     * Note that if this property returns `false`, it does not guarantee that consecutive call to [send] will succeed, as the
+     * channel can be concurrently closed right after the check. For such scenarios, it is recommended to use [trySend] instead.
+     *
```

Comment:
```suggestion
     *
     * @see SendChannel.trySend
```
to highlight what is actually meant in the majority of cases?

Also, WDYT about marking this API as delicate?

Reply:
> Also, WDYT about marking this API as delicate?

The question probably got lost.

Reply:
Not very keen on the idea, to be honest. I would prefer having `delicate` marker on a widespread or potentially really dangerous API, so it keeps having its weight.
I would like to avoid using it as "please just read a documentation" marker

Reply:
Sold!